File: qqbot-admin-api/src/main/java/com/mieai/qqbot/admin/bot/MediaUploadResponse.java
package com.mieai.qqbot.admin.bot;

import com.mieai.qqbot.client.MediaAsset;
import java.time.Instant;
import java.util.UUID;

public record MediaUploadResponse(UUID id, UUID botId, String kind, String fileName,
        String contentType, long sizeBytes, Instant createdAt) {
    static MediaUploadResponse from(MediaAsset asset) {
        return new MediaUploadResponse(asset.id(), asset.botId().value(), asset.kind().name(),
                asset.fileName(), asset.contentType(), asset.sizeBytes(), asset.createdAt());
    }
}

File: qqbot-module-qqbot-runtime/src/main/kotlin/com/mieai/qqbot/admin/bot/MediaUploadResponse.kt
package com.mieai.qqbot.admin.bot

import com.mieai.qqbot.client.MediaAsset
import java.time.Instant
import java.util.UUID

data class MediaUploadResponse(
    val id: UUID,
    val botId: UUID,
    val kind: String,
    val fileName: String,
    val contentType: String,
    val sizeBytes: Long,
    val createdAt: Instant,
) {
    companion object {
        fun from(asset: MediaAsset) = MediaUploadResponse(
            asset.id,
            asset.botId.value,
            asset.kind.name,
            asset.fileName,
            asset.contentType,
            asset.sizeBytes,
            asset.createdAt,
        )
    }
}
